src/components/ruleEditor/ruleEditor.js

```diff
@@ -168,14 +168,17 @@ class RuleEditor extends React.Component {
         devices.forEach(device => {
             const telemetry = device.Properties.Reported.Telemetry;
             if (telemetry) {
-                Object.keys(telemetry).forEach((field) => {
-                    const extract = field.match(/-(.*);/).pop();
-                    if (!fields.some((o) => o.value === extract)) {
-                        fields.push({
-                            label: extract,
-                            value: extract
-                        });
-                    }
+                Object.values(telemetry).forEach(field => {
+                    const extract = field.MessageSchema.Fields;
+                    Object.keys(extract).forEach(field => {
+                        if (field.indexOf('_') !== -1) return; //we don't want keys that contain _
```

Comment:
Why are we using the underscore convention to filter?  Are we sure variables we want in here will never have underscore?  This pushes a convention @ minimum to the Telemetry service, but it probably bleeds to telemetry-agent and simulation.

Reply:
![pasted image at 2017_10_13 11_31 am](https://user-images.githubusercontent.com/1734193/31560937-1e582bdc-b00b-11e7-870b-7890b502892e.png)

The underscore is being used to help filter the Fields object. As I understand it, we don't need the `_unit` keys. It appears that there is some convention already being used in the backend. Are there issues with this assumption?

Reply:
Yes. Devis also confirmed that, which ever fileds key is ends with '_unit' or value is text will be eliminated from the field dropdown.